lib/ecto/query/planner.ex

```diff
@@ -2027,15 +2027,24 @@ defmodule Ecto.Query.Planner do
   defp cte_fields([], [], _aliases), do: []
 
   defp assert_update!(%Ecto.Query{updates: updates} = query, operation) do
+    schema = schema_for_update(query)
```

Comment:
Call it `dumper_for_update` and skip one level of conditional? :)

```suggestion
    dumper = dumper_for_update(query)
```